src/fsharp/tast.fs

```diff
@@ -2401,18 +2422,27 @@ and [<StructuredFormatDisplay("{LogicalName}")>]
     member x.IsCompilerGenerated        = x.val_flags.IsCompilerGenerated
     
     /// Get the declared attributes for the value
-    member x.Attribs                    = x.val_attribs
+    member x.Attribs                    = 
+        match x.val_opt_data with
+        | Some x -> x.val_attribs
+        | _ -> []
 
     /// Get the declared documentation for the value
-    member x.XmlDoc                     = x.val_xmldoc
+    member x.XmlDoc                     =
+        match x.val_opt_data with
+        | Some x -> x.val_xmldoc
+        | _ -> XmlDoc.Empty
     
     ///Get the signature for the value's XML documentation
     member x.XmlDocSig 
-        with get() = x.val_xmldocsig
-        and set(v) = x.val_xmldocsig <- v
+        with get() = match x.val_opt_data with | Some x -> x.val_xmldocsig | _ -> String.Empty
+        and set(v) = match x.val_opt_data with | Some x -> x.val_xmldocsig <- v | _ -> x.val_opt_data <- Some { Val.EmptyValOptData with val_xmldocsig = v }
```

Comment:
Make this a multi-line expression.  There are just too many `<-`, `->` etc. on the same line :)